osu.Game/Graphics/UserInterface/ExtendableButton.cs

```diff
@@ -0,0 +1,117 @@
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Transformations;
+using OpenTK;
+using OpenTK.Graphics;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class ExtendableButton : ClickableContainer
+    {
+        public Color4 BGColour
+        {
+            get { return bgBox.Colour; }
+            set { bgBox.Colour = value; }
+        }
+        public new Color4 Colour
+        {
+            get { return textBox.Colour; }
+            set { textBox.Colour = value; }
+        }
+        public new Vector2 Shear
+        {
+            get { return textBox.Shear; }
+            set { textBox.Shear = value; }
+        }
+        public string Text
+        {
+            get { return spriteText?.Text; }
+            set
+            {
+                if (spriteText != null)
+                    spriteText.Text = value;
+            }
+        }
+
+        private Box bgBox;
+        private Box textBox;
+        public Container textContainer;
+        private SpriteText spriteText;
+
+        public Vector2 ExtendLenght = new Vector2(80,0); // this one can be defaulted
+        public Vector2 InitialExtendLenght/* = new Vector2(40, 0)*/; // but this one unfortunately cant
+
+        public ExtendableButton(float initialExtendLenghtX = 20, float initialExtendLenghtY = 0)
+        {
+            InitialExtendLenght = new Vector2(initialExtendLenghtX, initialExtendLenghtY); // because it just wont use defined settings when creating
+
+            Children = new Drawable[]
+            {
+                bgBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                },
+                textContainer = new Container
+                {
+                    Origin = Anchor.TopLeft,
+                    Anchor = Anchor.TopLeft,
+                    RelativeSizeAxes = Axes.Both,
+                    Position = Position + InitialExtendLenght,
+                    Children = new Drawable[]
+                    {
+                        textBox = new Box
+                        {
+                            Origin = Anchor.TopLeft,
+                            Anchor = Anchor.TopLeft,
+                            RelativeSizeAxes = Axes.Both,
+                            Shear = new Vector2(0.1f, 0), // should be relative?
+                            EdgeSmoothness = new Vector2(1, 1), // should be based on which side is being cut?
+                        },
+                        spriteText = new SpriteText
+                        {
+                            Origin = Anchor.Centre,
+                            Anchor = Anchor.Centre,
+                        }
+                    }
+                }
+            };
+        }
+
+        protected override bool OnHover(InputState state)
+        {
+            textContainer.ClearTransformations();
+            textContainer.MoveTo(Position + ExtendLenght, 150, EasingTypes.OutElastic);

```

Comment:
i get that, but in its current state it wouldn't really be useful (especially because it has a constant transition delay that is 100% correlating to the same delay values in BackButton).

i wouldn't bother extracting logic like this without actually *knowing* you have at least two use cases.